tests/components/totalconnect/test_alarm_control_panel.py

```diff
@@ -0,0 +1,85 @@
+"""The tests for the totalconnect alarm_control_panel platform."""
+from unittest.mock import Mock
+
+from homeassistant.components.totalconnect.alarm_control_panel import TotalConnectAlarm
+from homeassistant.const import (
+    STATE_ALARM_ARMED_AWAY,
+    STATE_ALARM_ARMED_CUSTOM_BYPASS,
+    STATE_ALARM_ARMED_HOME,
+    STATE_ALARM_ARMED_NIGHT,
+    STATE_ALARM_ARMING,
+    STATE_ALARM_DISARMED,
+    STATE_ALARM_DISARMING,
+    STATE_ALARM_TRIGGERED,
+)
+
+
+def test_alarm_update():
+    """Test if the alarm updates properly."""
+
+    mock_location = Mock()
+    mock_location.ac_loss = False
+    mock_location.low_battery = False
+    mock_location.is_cover_tampered.return_value = False
+
+    mock_client = Mock()
+    mock_client.locations = {}
+    mock_client.locations["123456"] = mock_location
+    mock_client.get_armed_status.return_value = True
+    mock_client.is_disarmed.return_value = True
+    mock_client.is_armed_home.return_value = True
+    mock_client.is_armed_night.return_value = True
+    mock_client.is_armed_away.return_value = True
+    mock_client.is_armed_custom_bypass.return_value = True
+    mock_client.is_arming.return_value = True
+    mock_client.is_disarming.return_value = True
+    mock_client.is_triggered_police.return_value = True
+    mock_client.is_triggered_fire.return_value = True
+    mock_client.is_triggered_gas.return_value = True
+
+    alarm = TotalConnectAlarm("test", "123456", mock_client)
```

Comment:
We should not instantiate the entity directly. Instead set up the component and platform let it add an entity and then assert the state of the entity via the state machine. Make sure to patch the library appropriately to avoid I/O.

See eg:
https://github.com/home-assistant/core/blob/e87fab6b5f9811beadf6ae101c2cb17933f4a235/tests/components/manual/test_alarm_control_panel.py#L85-L121